lib/offsite_payments/integrations/poli_pay.rb

```diff
@@ -0,0 +1,269 @@
+          result['NavigateURL']
+        rescue ActiveUtils::ResponseError => e
+          raise UrlRequestError, e
+        end
+
+        class UrlRequestError < RequestError
+          ERRORS = {
+            '14050' => "A transaction-specific error has occurred",
+            '14053' => "The amount specified exceeds the individual transaction limit set by the merchant",
+            '14054' => "The amount specified will cause the daily transaction limit to be exceeded",
+            '14055' => "General failure to initiate a transaction",
+            '14056' => "Error in merchant-defined data",
+            '14057' => "One or more values specified have failed a validation check",
+            '14058' => "The monetary amount specified is invalid",
+            '14059' => "A URL provided for one or more fields was not formatted correctly",
+            '14060' => "The currency code supplied is not supported by POLi or the specific merchant",
+            '14061' => "The MerchantReference field contains invalid characters",
+            '14062' => "One or more fields that are mandatory did not have values specified",
+            '14099' => "An unexpected error has occurred within transaction functionality"
+          }
+
+          def errors
+            ERRORS
+          end
+        end
+      end
+
+      class QueryInterface < Interface
+        def self.url(token)
+          "#{base_url}/Transaction/GetTransaction?token=#{CGI.escape(token)}"
+        end
+
+        def call(token)
+          raise ArgumentError, "Token must be specified" if token.blank?
+          raw_response = ssl_get(self.class.url(token), standard_headers)
+          parse_response(raw_response)
+        rescue ActiveUtils::ResponseError => e
+          raise QueryRequestError, e
+        end
+
+        class QueryRequestError < RequestError
+          ERRORS = {
+            '14050' => "Transaction was initiated by another merchant or another transaction-based error",
+            '14051' => "The transaction was not found",
+            '14052' => "The token provided was incomplete, corrupted or doesn't exist"
+          }
+
+          def errors
+            ERRORS
+          end
+        end
+      end
+
+      class FinancialInstitutionsInterface < Interface
+        def self.url
+          "#{base_url}/Entity/GetFinancialInstitutions"
+        end
+
+        def call
+          raw_response = ssl_get(self.class.url, standard_headers)
+          result = parse_response(raw_response)
+          result.map { |attr| FinancialInstitution.new(attr) }
+        end
+      end
+
+      class Helper < OffsitePayments::Helper
+        attr_reader :token_parameters
+
+        def initialize(order, account, options = {})
+          @login    = account
+          @password = options.fetch(:password)
+          @options  = options.except(:password).merge(order: order)
+          options.delete(:homepage_url) # must delete or super below crashes
```

Comment:
Why don't you simply pass ``options.except(:password, :homepage_url)`` to super instead of being destructive?